google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -1056,79 +1042,19 @@ public boolean equals(Object obj) {
     return obj == this
         || obj != null
             && obj.getClass().equals(BlobInfo.class)
-            && Objects.equals(toPb(), ((BlobInfo) obj).toPb());
+            && Objects.equals(
+                Conversions.apiary().blobInfo().encode(this),
+                Conversions.apiary()
+                    .blobInfo()
+                    .encode(((BlobInfo) obj))); // TODO: remove this excessive allocation
```

Comment:
When you say remove excessive allocation,  are you saying the Blob should retain a reference to it's encoding so you can don't have to do `.encode(this)`?

Reply:
No, I think we should implement equals & hash code in terms of the properties of the class and not in terms of it's converted result.